kernel/Kconfig

```diff
@@ -1138,6 +1138,28 @@ endif # BOOTARGS
 
 endmenu
 
+config KERNEL_NO_LTO
+	bool "Build selected kernel core files without LTO"
+	depends on LTO
+	help
+	  Some SoCs require kernel code to be placed in RAM, which makes link-time
+	  optimization (LTO) unsuitable for these files (-fno-lto). Disabling LTO
+	  allows the affected code to be linked as separate objects and placed in
+	  specific memory regions.
+
+	  Running kernel code from RAM can improve execution performance, especially
+	  for timing-critical routines or context switch paths.
+
+if KERNEL_NO_LTO
+config KERNEL_LTO_ALLOWLIST
+	string "List of kernel source files to keep LTO"
+	help
+	  List of kernel source filenames that should retain LTO even when
+	  CONFIG_KERNEL_NO_LTO is enabled.
+	  Example: CONFIG_KERNEL_LTO_ALLOWLIST="init.c errno.c fatal.c"
```

Comment:
Did you want to make these files the default setting?  Otherwise you could expand the help text to suggest this list as a good default as these features don't benefit from the performance gain of running in RAM.

Reply:
If we set default "init.c errno.c fatal.c" for CONFIG_KERNEL_LTO_ALLOWLIST, users would still need to override the entire string when adding more files. Kconfig does not support appending to the default value, so it cannot be extended incrementally.

Reply:
Sure, once this PR is merged, I’ll create a follow-up PR to enable.